### PR TITLE
docs: container and image related methods of containerEngine api

### PR DESCRIPTION
**Pull request by feloy**
Signed-off-by: Philippe Martin <phmartin@redhat.com>

### What does this PR do?

Document Container and Image-related methods on the containerEngine namespace of the API

More text formatting to come...

### What issues does this PR fix or reference?

Part of #5812

### CODE REVIEW

File: packages/extension-api/src/extension-api.d.ts [hunk 1]
```diff
@@ -2350,30 +2350,113 @@ declare module '@podman-desktop/api' {
...
   export namespace containerEngine {
+    /**
+     * Returns the list of containers across all container engines.
+     *
+     * @return A promise resolving to an array of containers information
+     */
     export function listContainers(): Promise<ContainerInfo[]>;
+
+    /**
+     * Get the Inspect information of a specific container, similar to what a `podman inspect` command returns
```
Comment (benoitf) [inline packages/extension-api/src/extension-api.d.ts]: I'm not sure to use 'podman inspect' as it's more behaving like a 'docker REST API' and then 'docker inspect' result

Comment (benoitf) [reply packages/extension-api/src/extension-api.d.ts]: maybe just avoid to use a specific engine name

File: packages/extension-api/src/extension-api.d.ts [hunk 2]
```diff
@@ -2382,12 +2465,34 @@ declare module '@podman-desktop/api' {
...
+    /**
+     * Delete a container image from a local engine
+     *
+     * @param engineId the id of the engine managing the image
+     * @param id the id of the image on this engine
+     */
     export function deleteImage(engineId: string, id: string): Promise<void>;
+
+    /**
+     * Get Inspect information for a local image, similarly to the `podman image inspect` command
```
Comment (benoitf) [inline packages/extension-api/src/extension-api.d.ts]: get rid of 'podman'

**Review by axel7083 — APPROVED** [review]
LGTM

File: packages/extension-api/src/extension-api.d.ts [hunk 3]
```diff
@@ -2350,30 +2350,113 @@ declare module '@podman-desktop/api' {
...
+     */
     export function buildImage(
       // build context directory
       context: string,
       eventCollect: (eventName: 'stream' | 'error' | 'finish', data: string) => void,
       options?: BuildImageOptions,
     );
+
+    /**
+     * Save a container image on disk, similarly to the `podman image save` command
```
Comment (benoitf) [inline packages/extension-api/src/extension-api.d.ts]: get rid of 'podman'

**Review by benoitf — COMMENTED** [review]
it may require some `{@link NameOfTheClass}`

many methods are self-explanatory so I would guess addition of typedoc would be to provide examples or explain stuff

for example an engineID, how to grab it or to know it

for id, that it can be the name of a container (or image or etc) or the sha or a partial sha, etc

**Comment by feloy** [comment]
> it may require some `{@link NameOfTheClass}`
> 
> many methods are self-explanatory so I would guess addition of typedoc would be to provide examples or explain stuff
> 
> for example an engineID, how to grab it or to know it
> 

I wanted to document this, but I'm not sure yet what is the way to have it. Only from listImages(), listContainers(), or is there a way to list the engines (I cannot find any)?


> for id, that it can be the name of a container (or image or etc) or the sha or a partial sha, etc

**Comment by benoitf** [comment]
> I wanted to document this, but I'm not sure yet what is the way to have it. Only from listImages(), listContainers(), or is there a way to list the engines (I cannot find any)?

yes it's from the list operations for now

probably let say something like for deleteContainer (could be fully reworked, just a sample)

`When getting all containers using the {@link engine.listContainers} function to enumerate all containers, you can retrieve the IDs and engine IDs of each container. Then you can use them as parameters to deleteContainer`

File: packages/extension-api/src/extension-api.d.ts [hunk 4]
```diff
@@ -2276,89 +2359,154 @@ declare module '@podman-desktop/api' {
...
+    /**
+     * Set the os/arch of the built image (and its base image, when using one) to the provided value instead of using the current operating system and architecture of the host
+     */
     platform?: string;
-    // Set the provider to use, if not we will try select the first one available (sorted in favor of Podman).
+
+    /**
+     * Set the provider to use, if not we will try select the first one available (sorted in favor of Podman)
+     */
     provider?: ProviderContainerConnectionInfo | containerDesktopAPI.ContainerProviderConnection;
```
Comment (feloy) [inline packages/extension-api/src/extension-api.d.ts:2380]: These 2 types are not declared on the file

Comment (benoitf) [reply packages/extension-api/src/extension-api.d.ts:2380]: I think only one should be there

**Review by benoitf — APPROVED** [review]
Thanks now the typedoc is providing real value 👏 and not only describing the name of the functions/parameters

File: packages/extension-api/src/extension-api.d.ts (hunk 4 above)
Comment (feloy) [reply packages/extension-api/src/extension-api.d.ts:2380]: `ContainerProviderConnection` is defined, but `containerDesktopAPI` makes it undiscoverable

Do you prefer I make the change as part of this PR? It should be this, right?

```
 provider?: ContainerProviderConnection;
```

Comment (benoitf) [reply packages/extension-api/src/extension-api.d.ts:2380]: I think it needs a separate PR
there is already one on this topic

Comment (benoitf) [reply packages/extension-api/src/extension-api.d.ts:2380]: https://github.com/containers/podman-desktop/pull/5880

**Pull request merged**